### PR TITLE
Endpoints Tab | 03 | Add test for store and sdk struct

### PR DESCRIPTION
**Pull request by bstavroulakis**
---
⚠️  _This branch depends on changes that will be introduced with #2420. For a list of changes please see: https://github.com/dcos/dcos-ui/compare/bill/feat/connection-endpoints/02-sdk-endpoint-store...bill/feat/connection-endpoints/03-sdk-tests-and-structs_ 

--- 

Add test for store and sdk endpoints struct that is returned from the store.

**Checklist**
- [ ] Did you add a JIRA issue in a commit message or as part of the branch name?
- [ ] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

### CODE REVIEW

**Comment by bstavroulakis** [comment]
Alfred test this

**Comment by bstavroulakis** [comment]
#2420

**Comment by bstavroulakis** [comment]
Alfreddo cappuccino test this

**Comment by orlandohohmeier** [comment]
Let's merge  #2420 and this one as it doesn't really make sense to review and ship them separately.

File: plugins/services/src/js/structs/SDKServiceEndpoint.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import Item from "#SRC/js/structs/Item";
+
+class SDKServiceEndpoint extends Item {
```
[the remark this answers is not in the document]
Comment (bstavroulakis) [reply plugins/services/src/js/structs/SDKServiceEndpoint.js]: No other struct has a constructor as I saw it so kept it consistent, no required fields are necessary I think for this

File: plugins/services/src/js/structs/SDKServiceEndpoint.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  getEndpointName() {
+    return this.get("endpointName");
+  }
+  getEndpointData() {
+    return this.get("endpointData");
+  }
+  getVip() {
+    const endpointData = this.get("endpointData");
+    if (!endpointData.vip) {
```
[the remark this answers is not in the document]
Comment (bstavroulakis) [reply plugins/services/src/js/structs/SDKServiceEndpoint.js]: nice

File: plugins/services/src/js/structs/SDKServiceEndpoint.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      return null;
```
[the remark this answers is not in the document]
Comment (bstavroulakis) [reply plugins/services/src/js/structs/SDKServiceEndpoint.js]: I kept it null so one could maybe use this in a comparison statement easily. But will change it to string.

File: plugins/services/src/js/structs/SDKServiceEndpoint.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    return Array.isArray(endpointData.vip)
+      ? endpointData.vip.join(", ")
+      : endpointData.vip;
+  }
+  getAddress() {
+    const endpointData = this.get("endpointData");
+    if (!endpointData.address) {
+      return null;
```
[the remark this answers is not in the document]
Comment (bstavroulakis) [reply plugins/services/src/js/structs/SDKServiceEndpoint.js]: I kept it null so one could maybe use this in a comparison statement easily. But will change it to string.

File: plugins/services/src/js/structs/SDKServiceEndpoint.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    return Array.isArray(endpointData.address)
+      ? endpointData.address.join(", ")
+      : endpointData.address;
+  }
+  getDns() {
+    const endpointData = this.get("endpointData");
+    if (!endpointData.dns) {
+      return null;
```
[the remark this answers is not in the document]
Comment (bstavroulakis) [reply plugins/services/src/js/structs/SDKServiceEndpoint.js]: I kept it null so one could maybe use this in a comparison statement easily. But will change it to string.

File: plugins/services/src/js/structs/__tests__/SDKServiceEndpoint-test.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+      endpointName: "core-site.xml",
+      endpointData: "<endpoint1>master.elastic.l4lb.thisdcos.directory:9200</endpoint1>"
+    });
+  });
+  describe("#getEndpointName", function() {
+    it("returns correct endpoint name", function() {
+      expect(this.endpointJSON.getEndpointName()).toEqual("master-http");
+      expect(this.endpointFile.getEndpointName()).toEqual("core-site.xml");
+    });
+  });
```
[the remark this answers is not in the document]
Comment (bstavroulakis) [reply plugins/services/src/js/structs/__tests__/SDKServiceEndpoint-test.js]: No required fields

**Comment by orlandohohmeier** [comment]
Closing as the respective changes are part of #2420

**Pull request closed without merging**